project_task_delegate/wizards/project_task_delegate.py

```diff
@@ -0,0 +1,125 @@
+             by the user to whom you delegate this task""")
+    state = fields.Selection([('pending', 'Pending'), ('done', 'Done'), ],
+                             'Validation State', default='pending',
+                             help="""New state of your own task. Pending will be
+             reopened automatically when the delegated task is closed""")
+
+    @api.onchange('project_id')
+    def onchange_project_id(self):
+        self.user_id = self.project_id.user_id.id
+
+    @api.model
+    def default_get(self, fields):
+        """
+        This function gets default values
+        """
+        res = super(ProjectTaskDelegate, self).default_get(fields)
+        context = dict(self._context or {})
+        record_id = context.get('active_ids', False)
+        task = self.env['project.task'].browse(record_id)
+        task_name = tools.ustr(task.name)
+
+        if 'project_id' in fields:
+            res['project_id'] = int(
+                task.project_id.id) if task.project_id else False
+
+        if 'name' in fields:
+            if task_name.startswith(_('CHECK: ')):
+                newname = tools.ustr(task_name).replace(_('CHECK: '), '')
+            else:
+                newname = tools.ustr(task_name or '')
+            res['name'] = newname
+        if 'planned_hours' in fields:
+            res['planned_hours'] = task.remaining_hours or 0.0
+        if 'prefix' in fields:
+            if task_name.startswith(_('CHECK: ')):
+                newname = tools.ustr(task_name).replace(_('CHECK: '), '')
+            else:
+                newname = tools.ustr(task_name or '')
+            prefix = _('CHECK: %s') % newname
+            res['prefix'] = prefix
+        if 'new_task_description' in fields:
+            res['new_task_description'] = task.description
+        return res
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        res = super(ProjectTaskDelegate, self).fields_view_get(
+            view_id=view_id, view_type=view_type, toolbar=toolbar,
+            submenu=submenu)
+        obj_tm = self.env['res.users'].browse(
+            self._uid).company_id.project_time_mode_id
+        tm = obj_tm and obj_tm.name or 'Hours'
+        if tm in ['Hours', 'Hour']:
+            return res
+
+        eview = etree.fromstring(res['arch'])
+
+        def _check_rec(eview):
+            if eview.attrib.get('widget', '') == 'float_time':
+                eview.set('widget', 'float')
+            for child in eview:
+                _check_rec(child)
+            return True
+
+        _check_rec(eview)
+        res['arch'] = etree.tostring(eview)
+        for field in res['fields']:
+            if 'Hours' in res['fields'][field]['string']:
+                res['fields'][field]['string'] = res['fields'][
+                    field]['string'].replace('Hours', tm)
+        return res
```

Comment:
AFAICS the point for this `fields_view_get()` is to make the "Hours" field visible, depending  on the value`user:id.company_id.project_time_mode_id`.
Wouldn't it be much simpler to have that value available as a related field, and then use it in an `attrs` attribute in the form view field?